packages/legend-extension-dsl-data-space/src/stores/query-builder/DataSpaceQueryBuilderState.ts

```diff
@@ -112,12 +113,11 @@ export const resolveUsableDataSpaceClasses = (
       (m) => m.info !== undefined,
     )
   ) {
-    const compatibleClassPaths =
-      mappingModelCoverageAnalysisResult.mappedEntities.map(
-        (e) => e.info?.classPath,
-      );
-    const uniqueCompatibleClasses = compatibleClassPaths.filter(
-      (val, index) => compatibleClassPaths.indexOf(val) === index,
+    const uniqueCompatibleClasses = uniq(
+      mappingModelCoverageAnalysisResult.mappedEntities
+        // is root entity filters out mapped classes
+        .filter((e) => e.info?.isRootEntity)
+        .map((e) => e.info?.classPath),
```

Comment:
will this handle classes mapped via different mappings? e.g. associationmapping

Reply:
yes, as association mappings would dictate if a specific property within a class is mapped. that class would be a root entity. we are looking for overall mapped classes not all mapped properties.